### PR TITLE
Changed String.format to String template

### PR DESCRIPTION
**Pull request by SanjaySargam**
## Pull Request template
Changed String.format to String template

## Fixes
Kotlin CleanUp #11449 

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt [hunk 1]
```diff
@@ -371,14 +371,14 @@ open class RobolectricTest : CollectionGetter {
...
-            ?: throw IllegalArgumentException(String.format("Could not find model '%s'", name))
+            ?: throw IllegalArgumentException("Could not find model '$name'")
         // PERF: if we modify newNote(), we can return the card and return a Pair<Note, Card> here.
         // Saves a database trip afterwards.
         val n = col.newNote(model)
         for ((i, field) in fields.withIndex()) {
             n.setField(i, field)
         }
-        check(col.addNote(n) != 0) { String.format("Could not add note: {%s}", fields.joinToString(separator = ", ")) }
+        check(col.addNote(n) != 0) { "Could not add note: {$fields.joinToString(separator = \", \")}" }
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt:381]: This is not correct and it doesn't replicate the original code. See your other change in this file, where you used the proper syntax with {}.

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 2]
```diff
@@ -101,7 +101,7 @@ class AnkiPackageExporterTest : RobolectricTest() {
         assertThat("Three files should exist", files, hasSize(3))
 
         // {"0":"filename.txt"}
-        val expected = String.format("{\"0\":\"%s\"}", tempFileInCollection.name)
+        val expected = "{\"0\":\"$tempFileInCollection.name\"}"
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:104]: This is not correct, please see the [documentation for kotlin string templates ](https://kotlinlang.org/docs/strings.html#string-templates)on how to handle expressions.

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt [hunk 3]
```diff
@@ -146,7 +146,7 @@ class AnkiPackageExporterTest : RobolectricTest() {
         temp.delete()
         val newFile = File(col.media.dir(), s)
         check(newFile.exists()) { "Could not create temp file" }
-        addNoteUsingBasicModel(String.format("<img src=\"%s\">", newFile.name), "Back")
+        addNoteUsingBasicModel("<img src=\"$newFile.name\">", "Back")
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:149]: This is not correct, please see the [documentation for kotlin string templates ](https://kotlinlang.org/docs/strings.html#string-templates)on how to handle expressions.

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt (hunk 1 above)
Comment (SanjaySargam) [reply AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt:381]: Ok got it

File: AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt (hunk 2 above)
Comment (SanjaySargam) [reply AnkiDroid/src/test/java/com/ichi2/libanki/AnkiPackageExporterTest.kt:104]: resolved

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt [hunk 4]
```diff
@@ -371,14 +371,14 @@ open class RobolectricTest : CollectionGetter {
...
-            ?: throw IllegalArgumentException(String.format("Could not find model '%s'", name))
+            ?: throw IllegalArgumentException("Could not find model '$name'")
         // PERF: if we modify newNote(), we can return the card and return a Pair<Note, Card> here.
         // Saves a database trip afterwards.
         val n = col.newNote(model)
         for ((i, field) in fields.withIndex()) {
             n.setField(i, field)
         }
-        check(col.addNote(n) != 0) { String.format("Could not add note: {%s}", fields.joinToString(separator = ", ")) }
+        check(col.addNote(n) != 0) { "Could not add note: ${fields.joinToString(separator = ", ")}" }
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt:381]: The original string wrapped the format identifier inside {}, so it should be something like:

`"Could not add note: {${fields.joinToString(separator = ", ")}}" }`

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt (hunk 4 above)
Comment (SanjaySargam) [reply AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt:381]: ok

**Review by lukstbit — APPROVED** [review]
Thanks!

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**